test/builtin/test_assignment.py

```diff
@@ -232,3 +232,9 @@ def test_set_and_clear_messages(str_expr, str_expected, message, out_msgs):
         failure_message=message,
         expected_messages=out_msgs,
     )
+
+
+def test_predecrement():
+    check_evaluation(
+        "--5", "4", failure_message="Set::setraw: Cannot assign to raw object 5."
```

Comment:
In `FullForm`: `PreIncrement[5]` should return unevaluated as `PreIncrement[5]` and print the error message.

Reply:
I made a small experimental branch to try to address this.  If this is wrong (and I don't pretend to understand WMA semantics), then I suspect we have wrong in a broader set of assignments where we don't roll back evaluation. 

Branch assignment-failure-keeps-lhs has some of this.